dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ResourcesService.java

```diff
@@ -1232,4 +1218,16 @@ void listAllChildren(int resourceId,List<Integer> childList){
         }
     }
 
+    /**
+     * save resource and  check save status
+     * @param resource resource
+     */
+    private void saveResource(Resource resource){
+        int saveResourceStatus = resourcesMapper.insert(resource);
+        if (saveResourceStatus != 1) {
+            logger.error("resource  maybe already exists, can't recreate ");
+            throw new DDLException("resource maybe already exists, can't recreate");
```

Comment:
Hi,
Is it better to use `ServiceException`, because I do not find the difference except name about `ServiceException` and `DDLException`, and I think this logic actually is service and business exception not ddl exception (ddl exception means the ddl statement process has exception).